### DISCUSSION TITLE
json_value_free and json_object_free

### DISCUSSION THREAD

**Discussion by OpenSourceEverything** [kgabis/parson#192] [Q&A] [answered]
Hello,

It is not clear to me whether a call to json_value_free and json_object_free are both needed when initializing your root value/object pair per example code in the readme.md.

In the example below, is the jason_value_free sufficient to free all memory in addition to the json_free_serialized_string call?

        JSON_Value  *root_value  = json_value_init_object();
        JSON_Object *root_object = json_value_get_object(root_value);
        
        json_object_set_string(root_object, "a","b");

        char *pBuff = json_serialize_to_string_pretty(root_value);
     
        json_free_serialized_string(pBuff);

        json_value_free(root_value);

**Comment by kgabis** [comment, marked as answer]
```json_object_free``` is not part of the API exposed to users. And yes, in the example above, ```json_value_free``` and ```json_free_serialized_string``` for the string are sufficient.
